dogpile/cache/region.py

```diff
@@ -1248,26 +1249,32 @@ def somemethod(self, x, y):
         if function_key_generator is None:
             function_key_generator = self.function_key_generator
 
-        def decorator(fn):
+        def get_or_create_wrapper(key_generator, user_func, *arg, **kw):
+            key = key_generator(*arg, **kw)
+
+            @wraps(user_func)
+            def creator():
+                return user_func(*arg, **kw)
+            timeout = expiration_time() if expiration_time_is_callable \
+                else expiration_time
+            return self.get_or_create(key, creator, timeout,
+                                      should_cache_fn)
+
+        def cache_decorator(user_func):
             if to_str is compat.string_type:
                 # backwards compatible
-                key_generator = function_key_generator(namespace, fn)
+                key_generator = function_key_generator(namespace, user_func)
             else:
                 key_generator = function_key_generator(
-                    namespace, fn,
+                    namespace, user_func,
                     to_str=to_str)
 
-            @wraps(fn)
-            def decorate(*arg, **kw):
+            # Like invalidate, but regenerates the value instead
```

Comment:
nit: I know this is a dynamically-defined function, but you might as well put this in the func docstring instead of a comment.